src/transformers/models/convnext_maskrcnn/modeling_convnext_maskrcnn.py

```diff
@@ -0,0 +1,3526 @@
+
+# Base docstring
+_CHECKPOINT_FOR_DOC = "facebook/convnext-tiny-maskrcnn"
+_EXPECTED_OUTPUT_SHAPE = [1, 768, 7, 7]
+
+
+CONVNEXTMASKRCNN_PRETRAINED_MODEL_ARCHIVE_LIST = [
+    "facebook/convnext-tiny-maskrcnn",
+    # See all ConvNextMaskRCNN models at https://huggingface.co/models?filter=convnext_maskrcnn
+]
+
+BYTES_PER_FLOAT = 4
+# TODO: This memory limit may be too much or too little. It would be better to
+# determine it based on available resources.
+GPU_MEM_LIMIT = 1024**3  # 1 GB memory limit
+
+
+@dataclass
+class MaskRCNNModelOutput(ModelOutput):
+    """
+    Base class for models that leverage the Mask R-CNN framework.
+
+    Args:
+        losses (...)
+            ...
+        results (...)
+            ...
+        last_hidden_state (`torch.FloatTensor` of shape `(batch_size, sequence_length, hidden_size)`):
+            Sequence of hidden-states at the output of the last layer of the model.
+        hidden_states (`tuple(torch.FloatTensor)`, *optional*, returned when `output_hidden_states=True` is passed or when `config.output_hidden_states=True`):
+            Tuple of `torch.FloatTensor` (one for the output of the embeddings, if the model has an embedding layer, +
+            one for the output of each layer) of shape `(batch_size, sequence_length, hidden_size)`. Hidden-states of
+            the model at the output of each layer plus the optional initial embedding outputs.
+        attentions (`tuple(torch.FloatTensor)`, *optional*, returned when `output_attentions=True` is passed or when `config.output_attentions=True`):
+            Tuple of `torch.FloatTensor` (one for each layer) of shape `(batch_size, num_heads, sequence_length,
+            sequence_length)`. Attentions weights after the attention softmax, used to compute the weighted average in
+            the self-attention heads.
+    """
+
+    losses: torch.FloatTensor = None
+    results: List[List[np.ndarray]] = None
+    last_hidden_state: torch.FloatTensor = None
+    hidden_states: Optional[Tuple[torch.FloatTensor]] = None
+    attentions: Optional[Tuple[torch.FloatTensor]] = None
+
+
+def multi_apply(func, *args, **kwargs):
+    """Apply function to a list of arguments.
+    Note:
+        This function applies the `func` to multiple inputs and map the multiple outputs of the `func` into different
+        list. Each list contains the same type of outputs corresponding to different inputs.
+    Args:
+        func (Function): A function that will be applied to a list of
+            arguments
+    Returns:
+        tuple(list): A tuple containing multiple list, each list contains \
+            a kind of returned results by the function
+    """
+    pfunc = partial(func, **kwargs) if kwargs else func
+    map_results = map(pfunc, *args)
+    return tuple(map(list, zip(*map_results)))
+
+
+def unmap(data, count, inds, fill=0):
+    """Unmap a subset of item (data) back to the original set of items (of size
+    count)"""
```

Comment:
Could you fill out this function a bit - types, docstring etc. It's not clear to me what is meant by "unmap" or the intended behaviour.